fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_BrakeTemp.js

```diff
@@ -20,186 +169,178 @@ const SPEED_COOLDOWN_FACTOR = 0.00055;
+        // Update simulation with new values
+        SimVar.SetSimVarValue(this.brakeTempSimVar, "celsius", this.brakeTemp);
+        SimVar.SetSimVarValue(this.reportedBrakeTempSimVar, "celsius", this.reportedBrakeTemp);
+    }
+}
+
+/**
+ * Main class handling all brakes
+ * It owns one GearStatus per landing gear, a BrakeFan system and 4 brakes.
+ */
+class A32NX_BrakeTemp {
 
-            const deltaHeatUpWheelSet1 = (_deltaTime / 1000) * this.calculateHeatUp(currentBrakeLeft, wheelSet1Rpm);
 
-            if (currentBrakeLeft > 0) {
-                const brakeHeatUp0 = this.getRandomArbitrary(0.5, 1.5) * deltaHeatUpWheelSet1;
-                const brakeHeatUp1 = this.getRandomArbitrary(0.5, 1.5) * deltaHeatUpWheelSet1;
-                currentBrakeTemps[0] += brakeHeatUp0;
-                currentBrakeTemps[1] += brakeHeatUp1;
-                currentReportedBrakeTemps[0] += brakeHeatUp0;
-                currentReportedBrakeTemps[1] += brakeHeatUp1;
-            }
+    constructor() {
+        this.initializedAmbientBrakeTemp = false;
+        this.gearLeft = new A32NX_GearStatus(0, "L:A32NX_GEAR_LEFT_POSITION", "BRAKE LEFT POSITION", "WHEEL RPM:1");
+        this.gearRight = new A32NX_GearStatus(1, "L:A32NX_GEAR_RIGHT_POSITION", "BRAKE RIGHT POSITION", "WHEEL RPM:2");
+        this.brakeFan = new A32NX_BrakeFan("L:A32NX_BRAKE_FAN", "L:A32NX_BRAKE_FAN_BTN_PRESSED", this.gearLeft)
+        this.brakes = [
+            new A32NX_OneBrakeTemp(0, "L:A32NX_BRAKE_TEMPERATURE_1", "L:A32NX_REPORTED_BRAKE_TEMPERATURE_1", this.gearLeft, this.brakeFan),
+            new A32NX_OneBrakeTemp(1, "L:A32NX_BRAKE_TEMPERATURE_2", "L:A32NX_REPORTED_BRAKE_TEMPERATURE_2", this.gearLeft, this.brakeFan),
+            new A32NX_OneBrakeTemp(2, "L:A32NX_BRAKE_TEMPERATURE_3", "L:A32NX_REPORTED_BRAKE_TEMPERATURE_3", this.gearLeft, this.brakeFan),
+            new A32NX_OneBrakeTemp(3, "L:A32NX_BRAKE_TEMPERATURE_4", "L:A32NX_REPORTED_BRAKE_TEMPERATURE_4", this.gearLeft, this.brakeFan)
+        ]
+    }
 
-            const deltaHeatUpWheelSet2 = (_deltaTime / 1000) * this.calculateHeatUp(currentBrakeRight, wheelSet2Rpm);
+    init() { }
 
-            if (currentBrakeRight > 0) {
-                const brakeHeatUp2 = this.getRandomArbitrary(0.5, 1.5) * deltaHeatUpWheelSet2;
-                const brakeHeatUp3 = this.getRandomArbitrary(0.5, 1.5) * deltaHeatUpWheelSet2;
-                currentBrakeTemps[2] += brakeHeatUp2;
-                currentBrakeTemps[3] += brakeHeatUp3;
-                currentReportedBrakeTemps[2] += brakeHeatUp2;
-                currentReportedBrakeTemps[3] += brakeHeatUp3;
-            }
+    /**
+     * Update loop
+     * Fail fast if simulation is not ready
+     * Get ambiant temperature and airspeed, then update gear/brake fan systems from simulation
+     * Then compute brake temperature evolution
+     * Finally compute if any brake can be considered HOT
+     * @param {number} _deltaTime
+     */
+    update(_deltaTime) {
```

Comment:
Note for this specific case as well as generally: We use underscores in front of argument names to indicate that the argument is not used in the function, i.e `_deltaTime`. In this case, it does appear to be used however, so I'd change the name to `deltaTime`.

I believe there are earlier occurences where the parameter is indeed unused. In those cases, I suggest you either remove the argument or change the name to `_deltaTime` :slight_smile: .